utils/exp_manager.py

```diff
@@ -747,7 +751,26 @@ def hyperparameters_optimization(self) -> None:
         )
 
         try:
-            study.optimize(self.objective, n_trials=self.n_trials, n_jobs=self.n_jobs)
+            if self.total_n_trials is not None:
+                counted_states = [
+                    TrialState.COMPLETE,
+                    TrialState.RUNNING,
+                    TrialState.PRUNED,
+                ]
+                completed_trials = sum(t.state in counted_states for t in study.trials)
```

Comment:
Do we want to count something as `TrialState.RUNNING` as completed? (Genuine question, not familiar with Optuna.)

Reply:
We count them as "potentially completed". If we do not count already running trials, then we get (total_n_trials + n_workers) many trials. Which obviously is too much and also is confusing.
The drawback is, that if a worker crashes and its trial is left in a RUNNING state, we have to delete the stale trial manually.
See also https://github.com/optuna/optuna/issues/3518 for details.